icu4c/source/tools/makeconv/genmbcs.cpp

```diff
@@ -173,7 +173,7 @@ MBCSOpen(UCMFile *ucm) {
     }
 
     MBCSInit(mbcsData, ucm);
-    return &mbcsData->newConverter;
+    return (NewConverter *)mbcsData;
```

Comment:
This change will break if someone
swap the order of 
```
    NewConverter newConverter;
    UCMFile *ucm;
```
in MBCSData right?

shouldn't changing it to
```
return &(mbcsData->newConverter);
```
be enough?

Reply:
The parentheses in `return &(mbcsData->newConverter);` are redundant.

Why this change?

Reply:
I thought it fixes this coverity warning:

```
Error: RESOURCE_LEAK (CWE-772): [#def25](https://unicode-org.atlassian.net/browse/ICU-21667#def25)
icu/source/tools/makeconv/genmbcs.cpp:169: alloc_fn: Storage is returned from allocation function "uprv_malloc_69".
icu/source/tools/makeconv/genmbcs.cpp:169: var_assign: Assigning: "mbcsData" = storage returned from "uprv_malloc_69(456528UL)".
icu/source/tools/makeconv/genmbcs.cpp:175: noescape: Resource "mbcsData" is not freed or pointed-to in "MBCSInit".
icu/source/tools/makeconv/genmbcs.cpp:176: leaked_storage: Variable "mbcsData" going out of scope leaks the storage it points to.

     174|   
     175|       MBCSInit(mbcsData, ucm);
     176|>     return &mbcsData>newConverter;
     177|   }
     178|
```

Reply:
Like this?:
```c++
index 774cdc6045d..a996ac5aa8d 100644
--- a/icu4c/source/tools/makeconv/genmbcs.cpp
+++ b/icu4c/source/tools/makeconv/genmbcs.cpp
@@ -173,6 +173,10 @@ MBCSOpen(UCMFile *ucm) {
     }
 
     MBCSInit(mbcsData, ucm);
+    /* The memory in the MBSData structure following
+     * newConverter will be properly freed in MBCSClose.
+     */
+    /* coverity[leaked_storage] */
     return &mbcsData->newConverter;
 }
```